internal/policies/apparmor/apparmor_test.go

```diff
@@ -0,0 +1,223 @@
+package apparmor_test
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/termie/go-shutil"
+	"github.com/ubuntu/adsys/internal/policies/apparmor"
+	"github.com/ubuntu/adsys/internal/policies/entry"
+	"github.com/ubuntu/adsys/internal/testutils"
+)
+
+var update bool
+
+func TestApplyPolicy(t *testing.T) {
+	t.Parallel()
+
+	defaultMachineProfile := []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo"}}
+
+	tests := map[string]struct {
+		entries []entry.Entry
+		user    bool
+
+		noParserOutput      bool
+		saveAssetsError     bool
+		apparmorParserError bool
+		destAlreadyExists   string
+		readOnlyApparmorDir string
+		noApparmorParser    bool
+
+		wantErr bool
+	}{
+		// computer cases
+		"computer, one profile":              {entries: defaultMachineProfile},
+		"computer, multiple profiles,":       {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo\nusr.bin.bar\nnested/usr.bin.baz"}}},
+		"computer, duplicated profiles":      {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo\nusr.bin.foo"}}},
+		"computer, blank line profiles":      {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo\n\nusr.bin.bar\n"}}},
+		"computer, only blank profiles":      {entries: []entry.Entry{{Key: "apparmor-machine", Value: "\n\n\n"}}},
+		"existing .old directory is removed": {entries: defaultMachineProfile, destAlreadyExists: "machine.old", noParserOutput: true},
+		"existing .new directory is removed": {entries: defaultMachineProfile, destAlreadyExists: "machine.new", noParserOutput: true},
+
+		// shared cases
+		"no profiles, existing rules are removed": {destAlreadyExists: "machine"},
```

Comment:
and I was going to write about the missing removal test with a pre-existing directory and I see that you did it. Well done :p (ofc, once we have the partial removal, you will handle this too)